earth/earth_test.go

```diff
@@ -0,0 +1,193 @@
+		if got, want := AngleFromLength(test.length), test.angle; !float64Eq(got.Radians(), want.Radians()) {
+			t.Errorf("AngleFromLength(%v) = %v, want %v", test.length, got, want)
+		}
+	}
+}
+
+func TestLengthFromAngle(t *testing.T) {
+	for _, test := range degreesToMeters {
+		if got, want := LengthFromAngle(test.angle), test.length; !float64Eq(got.Meters(), want.Meters()) {
+			t.Errorf("LengthFromAngle(%v) = %v, want %v", test.angle, got, want)
+		}
+	}
+}
+
+func TestLengthFromPoints(t *testing.T) {
+	tests := []struct {
+		x1, y1, z1 float64
+		x2, y2, z2 float64
+		length     unit.Length
+	}{
+		{1, 0, 0, 1, 0, 0, 0 * unit.Meter},
+		{1, 0, 0, 0, 1, 0, 10007559.105973555 * unit.Meter},
+		{1, 0, 0, 0, 1, 1, 10007559.105973555 * unit.Meter},
+		{1, 0, 0, -1, 0, 0, 20015118.21194711 * unit.Meter},
+		{1, 2, 3, 2, 3, -1, 7680820.247060414 * unit.Meter},
+	}
+	for _, test := range tests {
+		p1 := s2.PointFromCoords(test.x1, test.y1, test.z1)
+		p2 := s2.PointFromCoords(test.x2, test.y2, test.z2)
+		if got, want := LengthFromPoints(p1, p2), test.length; !float64Eq(got.Meters(), want.Meters()) {
+			t.Errorf("LengthFromPoints(%v, %v) = %v, want %v", p1, p2, got, want)
+		}
+	}
+}
+
+func TestLengthFromLatLngs(t *testing.T) {
+	tests := []struct {
+		lat1, lng1, lat2, lng2 float64
+		length                 unit.Length
+	}{
+		{90, 0, 90, 0, 0 * unit.Meter},
+		{-37, 25, -66, -155, 8562022.790666264 * unit.Meter},
+		{0, 165, 0, -80, 12787436.635410652 * unit.Meter},
+		{47, -127, -47, 53, 20015118.077688109 * unit.Meter},
+	}
+	for _, test := range tests {
+		ll1 := s2.LatLngFromDegrees(test.lat1, test.lng1)
+		ll2 := s2.LatLngFromDegrees(test.lat2, test.lng2)
+		if got, want := LengthFromLatLngs(ll1, ll2), test.length; !float64Eq(got.Meters(), want.Meters()) {
+			t.Errorf("LengthFromLatLngs(%v, %v) = %v, want %v", ll1, ll2, got, want)
+		}
+	}
+}
+
+func TestAreaFromSteradians(t *testing.T) {
+	const earthArea = 6371.01 * 6371.01 * math.Pi * 4 * unit.SquareKilometer
+	tests := []struct {
+		steradians float64
+		area       unit.Area
+	}{
+		{1, earthArea / 4 / math.Pi},
+		{s2.PolygonFromLoops([]*s2.Loop{s2.FullLoop()}).Area(), earthArea},
+		{s2.PolygonFromLoops([]*s2.Loop{s2.LoopFromPoints([]s2.Point{
+			s2.PointFromLatLng(s2.LatLngFromDegrees(-90, 0)),
+			s2.PointFromLatLng(s2.LatLngFromDegrees(0, 0)),
+			s2.PointFromLatLng(s2.LatLngFromDegrees(90, 0)),
+			s2.PointFromLatLng(s2.LatLngFromDegrees(0, -90)),
+		})}).Area(), earthArea / 4},
+		{s2.CellFromCellID(s2.CellIDFromFace(2)).ExactArea(), earthArea / 6},
+		{s2.AvgAreaMetric.Value(10), 81.07281893380302 * unit.SquareKilometer},  // average area of level 10 cells
+		{s2.AvgAreaMetric.Value(20), 77.31706517582228 * unit.SquareMeter},      // average area of level 20 cells
+		{s2.AvgAreaMetric.Value(30), 73.73529927808979 * unit.SquareMillimeter}, // average area of level 30 cells
+		{s2.PolygonFromLoops([]*s2.Loop{s2.EmptyLoop()}).Area(), 0 * unit.SquareMeter},
+	}
```

Comment:
The function mentions "The value will be between 0 and 4 * math.Pi if a does not exceed the area of the Earth." 

Should there be any sanity bounds checks in the function to ensure its capped at 4 pi?